nix/modules/devShells.nix

```diff
@@ -172,26 +170,34 @@
             '';
           };
 
-        rustDev =
-          mkRustShell
-            {
-              inputsFrom = [
-                self'.packages.holochain
-              ];
+        rustDev = pkgs.mkShell {
+          inputsFrom = [
+            self'.packages.holochain
+          ];
 
-              shellHook = ''
-                export CARGO_HOME="$PWD/.cargo"
-                export CARGO_INSTALL_ROOT="$PWD/.cargo"
-                export CARGO_TARGET_DIR="$PWD/target"
-                export CARGO_CACHE_RUSTC_INFO=1
-                export PATH="$CARGO_INSTALL_ROOT/bin:$PATH"
-                export NIX_PATH="nixpkgs=${pkgs.path}"
-                export PS1='\n\[\033[1;34m\][rustDev:\w]\$\[\033[0m\] '
-                echo Rust development shell spawned. Type 'exit' to leave.
-              '' + (lib.strings.optionalString pkgs.stdenv.isDarwin ''
-                export DYLD_FALLBACK_LIBRARY_PATH="$(rustc --print sysroot)/lib"
-              '');
-            };
+          packages = (lib.lists.optionals pkgs.stdenv.isLinux [
+            pkgs.mold
+            pkgs.pkgsStatic.openssl
+          ]);
+
+          shellHook = ''
+            export CARGO_HOME="$PWD/.cargo"
+            export CARGO_INSTALL_ROOT="$PWD/.cargo"
+            export CARGO_TARGET_DIR="$PWD/target"
+            export CARGO_CACHE_RUSTC_INFO=1
+            export PATH="$CARGO_INSTALL_ROOT/bin:$PATH"
+            export NIX_PATH="nixpkgs=${pkgs.path}"
+            export PS1='\n\[\033[1;34m\][rustDev:\w]\$\[\033[0m\] '
+            echo Rust development shell spawned. Type 'exit' to leave.
+          ''
+          + (lib.strings.optionalString pkgs.stdenv.isDarwin ''
+            export DYLD_FALLBACK_LIBRARY_PATH="$(rustc --print sysroot)/lib"
+          '')
+          + (lib.strings.optionalString pkgs.stdenv.isLinux ''
+            export RUSTFLAGS="$RUSTFLAGS -Clink-arg=-fuse-ld=mold"
```

Comment:
Does this inherit Rust flags from outside the Nix environment or is this just able to inherit flags set elsewhere by us?

Reply:
> Does this inherit Rust flags from outside the Nix environment

yes:

```
$ env RUSTFLAGS="outside" nix develop .#coreDev
Rust development shell spawned. Type exit to leave.
pre-commit-hooks.nix: hooks up to date

[coreDev:~/src/holo/holochain]$ echo $RUSTFLAGS
outside -Clink-arg=-fuse-ld=mold
```

Reply:
Okay cool, that's something to be aware of then. Thanks for demonstrating :)

Reply:
my rationale is that the user can prevent any environment variables from being passed into the `nix develop` shell:

```
      · --ignore-environment / -i

        Clear the entire environment (except those specified with --keep).
```

so keeping the possibility of actually using this one instead of clobbering it.